portfolio/src/main/webapp/script.js

```diff
@@ -16,44 +16,107 @@ window.onload = function() {
     
     fetch("/data?comment-capacity=" + commentCapacity).then(response => response.json()).then((commentsJson) => {
 
-        console.log(commentsJson);
         const commentSectionElement= document.getElementById("comments");
 
         for (comment of commentsJson) {
-            commentSectionElement.appendChild(createCommentElement(comment));
+            console.log(comment);
+            json = JSON.parse(comment);
+            commentSectionElement.appendChild(createcommentElementFromJson(json));
         }
     });
 }
 
-function createCommentElement(text) {
-    const pElement = document.createElement("p");
-    pElement.innerText = text;
-    return pElement;
+function createcommentElementFromJson(json) {
+    let nameElement = document.createElement("b");
+    let nameText = document.createTextNode(json.name + ": ");
+    nameElement.appendChild(nameText);
+    let commentText = document.createTextNode(json.comment);
+
+    let comment = document.createElement("p");
+    comment.appendChild(nameElement);
+    comment.appendChild(commentText);
+    return comment;
 }
 
 function deleteComments() {
-    console.log("fetching deletion");
     fetch("/delete-data", { method: "post" }).then(response => console.log(response.text()));
     getComments();
 }
 
 function printCommentSender() {
-    fetch("/Authenticate").then(userData => userData.json()).then(nicknameJson => {
-        console.log(nicknameJson);
+    fetch("/user-data").then(userData => userData.json()).then(Json => {
         commentSenderDiv = document.getElementById("comment-sender-div");
-        commentSenderDiv.innerHTML =
-        `<p>Posting as <b>${nicknameJson.nickname}</b>. visit <a class="hyperlink" href="/settings">Settings</a> to update nickname.</p>
-        <form id="comment-box" action="/data" method="POST">
-            <textarea name="comment-input" placeholder="Comment on this page"></textarea>
-            <input value="Send" type="submit">
-        </form>`;
+
+        if (Json.userLoggedIn) {
+            commentSenderJs(commentSenderDiv);
+            //commentSenderDiv.innerHTML = commentSender();
+            document.getElementById("nickname-space").innerText = Json.nickname;
+        } else {
+            commentSenderDiv.innerHTML = loginLink(Json.loginUrl);
+        }
+        
     });
+}
+
+function loginLink(loginUrl) {
+    linkHtml =
+    `<p><a class="hotpink" href="${loginUrl}">Login</a> to post a comment.</p>`;
+    return linkHtml;
+}
+
+function commentSender() {
+    senderHtml = 
+    `<p>Commenting as <span class="hotpink"><b id="nickname-space"></b></span>. Visit <a class="hotpink" href="/settings.html">Settings</a> to update nickname.</p>
+    <form id="comment-box" action="/data" method="POST">
+        <textarea name="comment-input" placeholder="Comment on this page"></textarea>
+        <input value="Send" type="submit">
+    </form>`;
+    return senderHtml;
+}
+
+function commentSenderJs(parentDiv) {
```

Comment:
Ah I see what you mean about the js making it much longer and less readable. Did you decide to add it to the html instead?